### ISSUE TITLE
unnecessary drawing when certain XEvents happen

### ISSUE THREAD

**Issue by matjam** [brndnmtthws/conky#512]
So I'm not super familiar with the Xlib stuff, but it looks like there is a lot of unnecessary calls happening to draw_stuff() and it should not set need_to_update in the X events loop; specifically lines 2237 - 2241 in conky.cc:

```c
#ifdef USE_ARGB
            if (!have_argb_visual) {
#endif
              if (ev.xproperty.atom == ATOM(_XROOTPMAP_ID) ||
                  ev.xproperty.atom == ATOM(_XROOTMAP_ID)) {
                // draw_stuff();
                // next_update_time = get_time();
                // need_to_update = 1;
              }
#ifdef USE_ARGB
```

If I comment out those lines (as commented), the problem goes away.

Configuration:

```
conky.config = {
    alignment = 'top_right',
    background = false,
    border_width = 1,
    cpu_avg_samples = 2,
    double_buffer = true,
	default_color = 'white',
    default_outline_color = 'white',
    default_shade_color = 'white',
    draw_borders = false,
    draw_graph_borders = true,
    draw_outline = false,
    draw_shades = false,
    use_xft = true,
    font = 'Hack:size=12',
    gap_x = 60,
    gap_y = 60,
    minimum_height = 5,
	minimum_width = 5,
    net_avg_samples = 2,
    no_buffers = true,
    out_to_console = false,
    out_to_stderr = false,
    extra_newline = false,
    own_window = true,
    own_window_class = 'Conky',
    own_window_type = 'override',
    own_window_argb_visual = true,
    own_window_transparent = true,
    stippled_borders = 0,
    update_interval = 1.0,
    uppercase = false,
    use_spacer = 'none',
    show_graph_scale = false,
    show_graph_range = false
}

conky.text = [[
${scroll 16 $nodename - $sysname $kernel on $machine | }
$hr
${color red}Uptime:$color $uptime
${color red}Frequency (in GHz):$color $freq_g
${color red}RAM Usage:$color $mem/$memmax - $memperc% ${membar 4}
${color red}Swap Usage:$color $swap/$swapmax - $swapperc% ${swapbar 4}
${color red}CPU Usage:$color $cpu% ${cpubar 4}
${color red}Processes:$color $processes  ${color grey}Running:$color $running_processes
$hr
${color red}File systems:
 / $color${fs_used /}/${fs_size /} ${fs_bar 6 /}
${color red}Networking:
Up:$color ${upspeed eth0} ${color grey} - Down:$color ${downspeed eth0}
$hr
${color red}Name              PID   CPU%   MEM%
${color lightgrey} ${top name 1} ${top pid 1} ${top cpu 1} ${top mem 1}
${color lightgrey} ${top name 2} ${top pid 2} ${top cpu 2} ${top mem 2}
${color lightgrey} ${top name 3} ${top pid 3} ${top cpu 3} ${top mem 3}
${color lightgrey} ${top name 4} ${top pid 4} ${top cpu 4} ${top mem 4}
]]
```
I'm running Ubuntu MATE 18.04 using the Nvidia binary driver under Xorg (not using Wayland).

I found that by installing variety, when it changes the wallpaper, it would trigger the redraws. But it also does it on other random events too.

**Comment by matjam** [comment]
So I made a little animated gif that demonstrates the issue.

Offscreen, I am activating and deactivating hipchat, which is sending some events to conky that's making it redraw faster than once per second.

![video of problem](https://thumbs.gfycat.com/AdmiredDistinctFlickertailsquirrel-max-14mb.gif)

I don't believe the fix I've done is a proper fix, as I do think certain events should result in a redraw in conky, such as if the wallpaper changes, but I don't know Xlib well so I can't suggest the correct fix here.

**Comment by lasers** [comment]
Closed via #789. Thank you for the report. Add `forced_redraw = false` to your `conky.config`.
